llama-index-core/llama_index/core/query_pipeline/query.py

```diff
@@ -744,146 +746,106 @@ def _validate_inputs(self, module_input_dict: Dict[str, Any]) -> None:
                 f"Input keys: {module_input_dict.keys()}\n"
             )
 
-    def _process_component_output(
+    def process_component_output(
```

Comment:
is this user-facing? when would they use this?

Reply:
I have greatly improved the UX here now, and added a notebook example